src/ice_shelf/MOM_ice_shelf.F90

```diff
@@ -2178,7 +2176,12 @@ subroutine initialize_ice_shelf(param_file, ocn_grid, Time, CS, diag, Time_init,
 
   call MOM_IS_diag_mediator_close_registration(CS%diag)
 
-  if (present(fluxes_in)) call initialize_ice_shelf_fluxes(CS, ocn_grid, US, fluxes_in)
+  if (present(fluxes_in)) then
+     call initialize_ice_shelf_fluxes(CS, ocn_grid, US, fluxes_in)
+     call register_restart_field(fluxes_in%shelf_sfc_mass_flux, "sfc_mass_flux", .true., CS%restart_CSp, &
```

Comment:
This restart registration is occurring after the call to `restore_state()` on about line 1900, so `fluxes_in%shelf_sfc_mass_flux` will not be read from the restart file.  Please consider moving this restart registration call earlier in the routine so that it occurs before the call to `restore_state()`.  Also, it might be worthwhile to verify that the restart writes and reads are both working as intended for this new variable, if this has not already been done.